### ISSUE TITLE
Index Checker creates `@LessThan(doubleValue)`

### ISSUE THREAD

**Issue by NhatDinh** [typetools/checker-framework#2030]
Consider:
```java
package math;

import java.math.RoundingMode;

public class UnparsableBug {
    double roundIntermediate(double x, RoundingMode mode){
        if (x <= 0.0) {
            return x;
        } else {
            return (long) x + 1;
        }

    }
    public static void main(String args[]) {

    }
}
```
Warning:
```java
Error:(15, 22) java: [expression.unparsable.type.invalid] Expression in dependent type annotation invalid: [error for expression: 0.0; error: '0']
```

**Comment by mernst** [comment]
This is related to the Index Checker.

@NhatDinh In the future, it will be helpful if you can give a complete file (not just a snippet) and an exact command line.

**Comment by NhatDinh** [comment]
While annotating` DoubleMath#roundIntermediate()`, this issue came up. 
Please consider this simplified version of the code:
```java
public class bug {
	double roundIntermediate(double x){
		if (x >= 0.0) {
			return x;
		} else {
			return (long) x - 1;
		}
	}
}
```
Command line to run this code:
```java
javacheck -processor index bug.java
```
Expected errors:
```java
Error:(6, 33) java: [expression.unparsable.type.invalid] Expression in dependent type annotation invalid: [error for expression: 0.0; error: '0']
```

**Comment by NhatDinh** [comment]
@smillst Please take another look.

**Comment by mernst** [comment]
@NhatDinh Why is the output that you provided expected?

I get the following output:
```
bug.java:6: warning: [cast.unsafe] "@LessThan("0.0") double" may not be casted to the type "@LessThan("[error for expression: 0.0; error: '0']") long"
			return (long) x - 1;
			       ^
bug.java:6: error: [expression.unparsable.type.invalid] Expression in dependent type annotation invalid: [error for expression: 0.0; error: '0']
			return (long) x - 1;
			        ^
1 error
1 warning
```

If this is a problem with expression parsing, we might want to wait until https://github.com/typetools/checker-framework/pull/2062 is merged.

**Comment by NhatDinh** [comment]
@mernst Sorry for the confusion. The expected output means what I've produced in my error log.

**Comment by mernst** [comment]
The root cause is that the Index Checker is creating an annotation `@LessThan("0.0")`.
Floating-point values are not legal for `@LessThan`.  Fixing this will eliminate the error.

The proximate cause is that the flow expression parser cannot parse double values.  There is no need to fix that yet.
